### PR TITLE
Renaming (cont.)

### PR DESCRIPTION
**Pull request by mforets**
This is a continuation of https://github.com/JuliaIntervals/AffineArithmetic.jl/pull/12, with the necessary changes such that the tests pass. In particular, 

- Added the method `range(::AffineArithmetic)` (which arguably can be used instead of `interval`, or the other way round),
- Overloaded the functions `eltype`, `zero` and `one` for elements of type `Affine`.
- Updated the project file with dependency upper bounds (useful for a future registration of this package).

Also, I saw that some of the range bounds improved, see the tests "Small powers and range".

### CODE REVIEW

**Comment by mforets** [comment]
Regarding the example in https://github.com/JuliaIntervals/AffineArithmetic.jl/issues/14, i get

```julia
julia> Affine(-1..1) + Affine(2020.0) + (Affine(1e23) - Affine(1e23))
affine=⟨2020.0; [1.0]; [0, 0]⟩; range=[2019, 2021]

julia> Affine(-1..1) + Affine(2020.0) + Affine(1e23) - Affine(1e23)
affine=⟨0.0; [1.0]; [0, 0]⟩; range=[0, 1]
```
Not sure why..

**Comment by dpsanders** [comment]
Thanks a lot for doing this @mforets!

The reason it's still failing is that even with the changes in this PR, the method is still not actually rigorously bounding all errors. Basically we need to go through and make everything into an interval internally to do this.

In this example we are still seeing the usual problems with floating-point numbers that differ by too many orders of magnitude.

File: src/affine.jl [hunk 1]
```diff
@@ -17,7 +17,14 @@ struct Affine{N,T<:AbstractFloat}
 end
 
 interval(x::Affine) = x.range
+range(x::Affine) = x.range
```
Comment (dpsanders) [inline src/affine.jl]: The reason I stopped using `range` is because `range` now means something very different in Base.

**Comment by dpsanders** [comment]
I think we should just go ahead and merge this. Do you agree @mforets? If so, please feel free!

File: src/affine.jl (hunk 1 above)
Comment (mforets) [reply src/affine.jl]: Ok, i opened https://github.com/JuliaIntervals/AffineArithmetic.jl/issues/17.

**Comment by mforets** [comment]
> Basically we need to go through and make everything into an interval internally to do this.

Got it. I agree to do this in a follow up PR; this one already contains many changes :)

> I think we should just go ahead and merge this.

I don't have rights to merge, but i agree!

**Comment by dpsanders** [comment]
Thanks!

**Pull request merged**